docs/metrics_provider.md

```diff
@@ -1,40 +1,46 @@
 # Metrics provider
 
-The idea about metrics provider is to have a metrics driven approach of
-verifying the operators functionality. An operator reconciles a system where
-several resources exist. As a means of measurement and as a safety net, it is a good
-practise to implement metrics providers that emit metrics about the managed
-system and its resources.
+The idea about metrics providers is to have a metrics driven approach of
+verifying the operators functionality. An operator constantly reconciles the
+current state of the system towards a more desired state eventually. As a means
+of measurement and as a safety net, it is a good practise to implement metrics
+providers that emit metrics about the managed system and its resources. The
```

Comment:
🇺🇸  nit but could we have a separate paragraph for "The collector implementation should be separate"?

I think that will be easier to read quickly and emphasizes this is something we've learnt over time.

Reply:
Please check again if that works for you now.

Reply:
Thanks LGTM.